# import statements

In [1]:
import pandas as pd
import numpy as np

# code to read student.csv into pandas dataframe 

In [2]:
student = pd.read_csv('student.csv', delimiter = ';')
student.columns

Index(['school', 'sex', 'age', 'address', 'famsize', 'Pstatus', 'Medu', 'Fedu',
       'Mjob', 'Fjob', 'reason', 'guardian', 'traveltime', 'studytime',
       'failures', 'schoolsup', 'famsup', 'paid', 'activities', 'nursery',
       'higher', 'internet', 'romantic', 'famrel', 'freetime', 'goout', 'Dalc',
       'Walc', 'health', 'absences', 'G1', 'G2', 'G3'],
      dtype='object')

# helper functions (if needed)

In [3]:
#python code...

# Python code to answer question 1) What is the minimum, maximum, average, and median final grade (G3 column)?

In [4]:
columnG3 = student['G3']
minG3 = np.min(columnG3)
print("min: ", minG3)
maxG3 = np.max(columnG3)
print("max: ", maxG3)
avgG3 = np.mean(columnG3)
print("average: ", avgG3)
medianG3 = np.median(columnG3)
print("median: ", medianG3)

min:  0
max:  20
average:  10.415189873417722
median:  11.0


# Text answer for question 1.

minimum: 0
maximum: 20
average: 10.415189873417722
median: 11.0

# Python code to answer question 2) Which subgroup of sex, male or female, had the highest average final grade (G3)?

In [5]:
female = (student['sex'] == 'F')
final_f = student.loc[female, 'G3']
avgFFG = np.mean(final_f)
print("female avg: ", avgFFG)

male = (student['sex'] == 'M')
final_m = student.loc[male, 'G3']
avgMFG = np.mean(final_m)
print("male avg: ", avgMFG)

female avg:  9.966346153846153
male avg:  10.914438502673796


# Text answer for question 2.  

Male

# Python code to answer question 3) Which subgroup of Fjob had the lowest average final grade?

In [6]:
at_home = (student['Fjob'] == 'at_home')
final_atHome = student.loc[at_home, 'G3']
avgAhFG = np.mean(final_atHome)
print("at home avg: ", avgAhFG)

health = (student['Fjob'] == 'health')
final_health = student.loc[health, 'G3']
avgHFG = np.mean(final_health)
print("health avg: ", avgHFG)

other = (student['Fjob'] == 'other')
final_other = student.loc[other, 'G3']
avgOFG = np.mean(final_other)
print("other avg: ", avgOFG)

services = (student['Fjob'] == 'services')
final_services = student.loc[services, 'G3']
avgSFG = np.mean(final_services)
print("services avg: ", avgSFG)

teacher = (student['Fjob'] == 'teacher')
final_teacher = student.loc[teacher, 'G3']
avgTFG = np.mean(final_teacher)
print("teacher avg: ", avgTFG)

at home avg:  10.15
health avg:  11.61111111111111
other avg:  10.193548387096774
services avg:  10.297297297297296
teacher avg:  11.96551724137931


# Text answer for question 3.

at_home Fjob lowest average final grade

# Python code to answer question 4) On average, did students in romantic relationships get higher or lower final grades?

In [7]:
relationship = (student['romantic'] == 'yes')
relationStatus = student.loc[relationship, 'G3']
avgRrFG = np.mean(relationStatus)
print("romanic relationship avg: ", avgRrFG)

alone = (student['romantic'] == 'no')
aloneAffect = student.loc[alone, 'G3']
avgAFG = np.mean(aloneAffect)
print("no romantic relationship avg: ", avgAFG)

romanic relationship avg:  9.575757575757576
no romantic relationship avg:  10.836501901140684


# Text answer for question 4.

lower

# Python code to answer question 5) On average, do people with internet access at home make higher or lower final grades than people without internet? 

In [8]:
internet = (student['internet'] == 'yes')
internetStatus = student.loc[internet, 'G3']
avgIFG = np.mean(internetStatus)
print("internet access avg: ", avgIFG)

no_internet = (student['internet'] == 'no')
internetState = student.loc[no_internet, 'G3']
avgNiFG = np.mean(internetState)
print("No internet access avg: ", avgNiFG)

internet access avg:  10.617021276595745
No internet access avg:  9.409090909090908


# Text answer for question 5.

higher

# Python code to answer question 6) What percentage of male students are in romantic relationships?  What percentage of female students are in romantic relationships?

In [20]:
mRomantic = np.where((student['romantic']== 'yes') & (student['sex']=='M'))
print(mromantic)
men = student.loc[student['sex'] == 'M']
percentM = (np.shape(mRomantic)[0]/np.shape(men)[0])*100
print(percentM)

females = student['sex'] == 'F'
fRomantic = student[relationship][females]
women = student.loc[student['sex'] == 'F']
percentF = (np.shape(fRomantic)[0]/np.shape(women)[0])*100
print(percentF)

0.53475935828877
37.980769230769226


C:\Users\weste\AppData\Local\Temp/ipykernel_14952/944344293.py:7: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  fRomantic = student[relationship][females]


# Text answer for question 6.

The percent of male students in a romantic relationship is 28.34%.
The percent of females in a romantic relationship is 37.98%.

# Python code to answer question 7) What is the correlation between freetime and final grade? 

In [10]:
columnFt = student['freetime']
cor_ftFG = np.corrcoef(columnFt, columnG3)
print(cor_ftFG)

[[1.         0.01130724]
 [0.01130724 1.        ]]


# Text answer for question 7.

0.01130724

# Python code to answer question 8) What is the correlation between studytime and final grade? 

In [11]:
columnSt = student['studytime']
cor_stFG = np.corrcoef(columnSt, columnG3)
print(cor_stFG)

[[1.         0.09781969]
 [0.09781969 1.        ]]


# Text answer for question 8.

0.09781969

# Python code to answer question 9) What is the correlation between studytime and freetime? 

In [12]:
cor_stFT = np.corrcoef(columnSt, columnFt)
print(cor_stFT)

[[ 1.         -0.14319841]
 [-0.14319841  1.        ]]


# Text answer for question 9.

-0.14319841

# Python code to answer question 10) Excluding columns G1 and G2, which numerical column had the biggest correlation with final grade (G3)?

In [17]:
numColumns = student.select_dtypes(include='int')
finalNumColumns = numColumns.drop(['G1', 'G2', 'G3'], axis = 1)

for column, data in finalNumColumns.items():
    corellation = np.corrcoef(data, columnG3)
    print(column, " corellation: ")
    print(corellation)

age  corellation: 
[[ 1.         -0.16157944]
 [-0.16157944  1.        ]]
Medu  corellation: 
[[1.        0.2171475]
 [0.2171475 1.       ]]
Fedu  corellation: 
[[1.         0.15245694]
 [0.15245694 1.        ]]
traveltime  corellation: 
[[ 1.         -0.11714205]
 [-0.11714205  1.        ]]
studytime  corellation: 
[[1.         0.09781969]
 [0.09781969 1.        ]]
failures  corellation: 
[[ 1.         -0.36041494]
 [-0.36041494  1.        ]]
famrel  corellation: 
[[1.         0.05136343]
 [0.05136343 1.        ]]
freetime  corellation: 
[[1.         0.01130724]
 [0.01130724 1.        ]]
goout  corellation: 
[[ 1.         -0.13279147]
 [-0.13279147  1.        ]]
Dalc  corellation: 
[[ 1.         -0.05466004]
 [-0.05466004  1.        ]]
Walc  corellation: 
[[ 1.         -0.05193932]
 [-0.05193932  1.        ]]
health  corellation: 
[[ 1.        -0.0613346]
 [-0.0613346  1.       ]]
absences  corellation: 
[[1.         0.03424732]
 [0.03424732 1.        ]]


# Text answer for question 10.  Please identify the column and correlation.

text...